oml/retrieval/retrieval_results.py

```diff
@@ -125,13 +143,18 @@ def visualize(
         if not isinstance(dataset, IQueryGalleryDataset):
             raise TypeError(f"Dataset has to support {IQueryGalleryDataset.__name__}. Got {type(dataset)}.")
 
+        nq1, nq2 = len(self.retrieved_ids), len(dataset.get_query_ids())
+        if nq1 != nq2:
+            raise RuntimeError(f"Number of queries in RetrievalResults and Dataset must match: {nq1} != {nq2}")
```

Comment:
Extract names of classes programmatically as above

Reply:
`type(dataset)` will show you full path to class, it's not friendly
`dataset.__class__.__name__` shows only name
and 
`self.__class__.__name__` for RetrievalResults

Reply:
done, thx!